mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBIndexesDefiner.java

```diff
@@ -147,11 +147,36 @@ void validateIndexSpecsForIndexes(Table table, IndexOptions indexOptions, Set<St
 			MongoDBIndexSpec indexSpec = new MongoDBIndexSpec( index,
 					getIndexOptionDocument( table, indexOptions.getOptionForIndex( index.getName() ) ) );
 			if ( validateIndexSpec( indexSpec ) ) {
-				indexSpecs.add( indexSpec );
+				addHandleDuplications( indexSpec );
 			}
 		}
 	}
 
+	private void addHandleDuplications(MongoDBIndexSpec indexSpec) {
+		// we can add any index defined on different collection or different key set
+		// we rely on hashcode and equals to verify conflicts
+		if ( !indexSpecs.contains( indexSpec ) ) {
+			indexSpecs.add( indexSpec );
+			return;
+		}
+
+		int existingIndexPosition = indexSpecs.indexOf( indexSpec );
+		MongoDBIndexSpec existingIndex = indexSpecs.get( existingIndexPosition );
+
+		// in case on conflict
+		// we rely on compareTo to choose the best candidate
```

Comment:
This comment doesn't add anything useful to the follow up code. Also. what's the "best" candidate?
I think you can remove and add a  function with a sensible name instead.

Reply:
What do you think of:
`public boolean hasPreferableConfgiruationOver(MongoDBIndexSpec other)`?

Reply:
I think the problem for me is that I don't understand how one configuration is preferable to the other.
If they both add an index why do you update it? Couldn't you simply keep the existing one?

Reply:
Suppose we have a unique constraint an a normal index non unique, like the test case. In this case I prefer to keep the unique property on the index, because probably the user want the unique.

Reply:
Then the configuration with unique it should take the precedence over the non unique one.

Reply:
I would expect the code to look something like:
```
if (isNotUnique(existingContraint)) {
    replaceWithUniqueConstratint(existingConstraint, newConstraint);
}
else {
   // keep existing (maybe)
}
```
But I'm not sure we should log a warning every time, like, if a user define a text index and a spatial index on the same field, shouldn't we throw an exception?
What do you think?

Reply:
The idea was to define a method to compare and choose between two configurations, but you're right maybe it is overdesign. In fact the only criteria that we have at the moment is only the uniqueness.  I'm going to simplify it.

Reply:
I have just remembered that there is another problem: the order of arrival of the configuration items is not guaranteed. Choosing different configurations over different creation can create a problem of consistency of the behaviour. For this reason I would like to keep the method to choose between configurations.